File: JavaScript/7js 2/FilterString.js
/*
|--------------------------------------------------------------------------
| Filter the number
|--------------------------------------------------------------------------
|
*/

function FilterString(value) {
   let res = [];
   for (let i = 0; i < value.length; i++) {
      const num = Number(value[i]);
      if (num / 1 == num) {
         res.push(num);
      }
   }
   return Number(res.join(""));
}


const FilterString = v => Number(v.replace(/\D/g, ''));


function FilterString(value) {
   return +value.split('').filter(n => !isNaN(n)).join('');
}


console.log(FilterString("123"), 123, 'Just return the numbers');
console.log(FilterString("a1b2c3"), 123, 'Just return the numbers');
console.log(FilterString("aa1bb2cc3dd"), 123, 'Just return the numbers');

File: Python/7py 2/filter_string.py
"""
|--------------------------------------------------------------------------
| Filter the number
|--------------------------------------------------------------------------
|
"""

def filter_string(string):
   return int(''.join(list(filter(lambda i: i.isnumeric() , string))))


def filter_string(string):
   return int(filter(str.isdigit, string))


def filter_string(str):
   return int(''.join(ch for ch in str if ch.isdigit()))


from re import findall
def filter_string(string: str) -> int:
   return int("".join(findall("\d", string)))


def filter_string(string):
   return int("".join([num for num in string if num in "0123456789"]))


print(filter_string("123"), 123, 'Just return the numbers')
print(filter_string("a1b2c3"), 123, 'Just return the numbers')
print(filter_string("aa1bb2cc3dd"), 123, 'Just return the numbers')
print(filter_string("aa 112 3dd"), 1123, 'Just return the numbers')
print(filter_string("11bb2c23c3"), 112233, 'Just return the numbers')